src/liquid/components/ld-header/ld-header.tsx

```diff
@@ -0,0 +1,136 @@
+import { Component, Host, h, Prop, Element, Watch } from '@stencil/core'
+import { getClassNames } from 'src/liquid/utils/getClassNames'
+
+/**
+ * @virtualProp ref - reference to component
+ * @virtualProp {string | number} key - for tracking the node's identity when working with lists
+ * @part container - Actual header element that limits the width of the header content
+ * @part logo - `ld-icon` element containing the default logo
+ * @part site-name - `ld-typo` element containing the site name
+ */
+@Component({
+  assetsDirs: ['assets'],
+  tag: 'ld-header',
+  styleUrl: 'ld-header.css',
+  shadow: true,
+})
+export class LdHeader {
+  @Element() el: HTMLElement
+  private lastOffset?: number
+  private currentHeight?: number
+
+  /** Hides header. */
+  @Prop({ mutable: true }) hidden = false
+
+  /** Hide the header when the user scrolls down and show it again, when the user scrolls up. */
+  @Prop() hideOnScroll = false
+
+  /** Title attribute of the logo link. */
+  @Prop() logoTitle?: string
+
+  /** URL that the logo links to. */
+  @Prop() logoUrl?: string
+
+  /** Make the header sticky. */
+  @Prop() sticky = false
+
+  /** Name shown on the right side of the logo. */
+  @Prop() siteName?: string
+
+  private updateScrollDirection = () => {
+    const offset = window.pageYOffset ?? document.documentElement.scrollTop
+
+    if (window.innerHeight + offset >= document.body.offsetHeight) {
+      this.hidden = false
+    } else if (offset > this.lastOffset && offset > this.currentHeight) {
+      this.hidden = true
+    } else {
+      this.hidden = false
+    }
+
+    // For mobile or negative scrolling
+    this.lastOffset = offset < 0 ? 0 : offset
+  }
+
+  @Watch('hideOnScroll')
+  connectedCallback() {
+    if (this.hideOnScroll) {
+      this.lastOffset = window.pageYOffset || document.documentElement.scrollTop
+      window.addEventListener('scroll', this.updateScrollDirection, false)
+    } else {
+      this.disconnectedCallback()
+    }
+  }
+
+  disconnectedCallback() {
+    window.removeEventListener('scroll', this.updateScrollDirection, false)
+  }
+
+  componentDidLoad() {
+    const ldMenu = this.el.querySelector('ld-menu')
+
+    if (ldMenu) {
+      ldMenu.orientation = 'horizontal'
+    }
+
+    this.currentHeight = this.el.getBoundingClientRect().height
+
+    this.el
```

Comment:
I still think it is not the responsibility of the ld-header component to resize icons within buttons. If one needs a different icon size, it should be possible to set the icon size via a prop either on the icon or the button component. In short, this feature may come as a surprise to anyone using the ld-header. Either we remove it or we document this behaviour explicitly.

Reply:
I guess this is a matter of opinion. I can explicitly mention it, but I think this goes into too much detail for the docs already. IMHO it should not be a problem to make the header take care of special requirements that only occur inside the header.

We can also move that part of the logic to the `ld-button`. What do you think about that? Would ask Nic about this, anyways.

Reply:
Yes, let's ask Nic. There should not be too much difference in how he uses the components within Figma from how we use them in our code. If he uses a "special button" where the icon is larger than the usual, then we need to implement such a "special button".
What I am afraid of is, that we do things automatically here, without telling anyone in the docs about it. Unexpected things happen as a result. Instead, the developer should initiate changes in layout explicitly, so that it is obvious what's happening.

Reply:
As agreed upon, I moved the icon resize to the `ld-button` component and explicitly mentioned the changes that the `ld-header` applies to its children in the docs.